main.cpp

```diff
@@ -131,233 +169,381 @@ struct CarWash
+        }
     };
 
-    //3 things it can do:
-    //    - assign button functionality
     void assignButtonFunctionality();
-    //    - delay compensation    // Calibrates control to combat latency
     float delayCompensation(float timeButtonPressed, float timeActionExecuted, JoyStick activeJoystick);
-    //    - adjust distance traveled    // Defines amount of movement needed to get from point A to point B based on amount of joystick movement
-    double adjustDistanceTraveled(int xPointA, int yPointA, int xPointB, int yPointB, JoyStick activeJoystick);
+    double adjustDistanceTraveled(int adjustmentAmount, double xPointA, double yPointA, double xPointB, double yPointB, JoyStick activeJoystick);
+    void storeItemInArcadeBox(){}
 };
 
-struct ArcadeBox 
+double Controls::adjustDistanceTraveled(int adjustedCursorSpeed, double xPointA, double yPointA, double xPointB, double yPointB, JoyStick activeJoystick)
 {
-    //5 properties:
-    //    - storage location (int)
-    int storageLocation = 0;
-    //    - number of openings (int)
-    int numOpenings = 4;
-    //    - weight (float)
-    float weight = 123.f;
-    //    - style (std::string)
-    std::string style = "Upright";
-    //    - number of panels (int)
-    int numPanels = 5;
-    //3 things it can do:
-    //    - store item here 
-    void storeItemHere(std::string storeableItem);
-    //    - open arcade box back panel
-    void openArcadeBoxBackPanel();
-    //    - disassemblable and transport
-    void disassembleAndTransport();
-};
+    double totalDifference = fabs((xPointB - xPointA) - (yPointB - yPointA));
+    
+    int activeCursorSpeed = activeJoystick.getActiveCursorSpeed();
+
+    return totalDifference *= (1 + (adjustedCursorSpeed - activeCursorSpeed));
+}
+
+// 
+// UDT No. 8
+// 
 
 struct Speakers 
 {
-    //5 properties:
-    //    - volume (float)
     float volume = 9.f;
-    //    - input device (int)
     int inputDevice = 0;
-    //    - left speaker output (double)
-    double SpeakerL = 5.0;
-    //    - right speaker output (double)
-    double SpeakerR = 5.0;
-    //    - mono audio (int)
+    double SpeakerLOutput = 5.0;
+    double SpeakerROutput = 5.0;
     int monoAudio = 0;
-    //3 things it can do:
-    //    - adjust volume    // Difference in initial volume setting and volume end value
+    
     double adjustVolume(double volumeKnobDiff);
-    //    - change input device
     void changeInputDevice();
-    //    - combine sound     // volume output mono-audio
     double combineSound(double SpeakerL, double SpeakerR);
+    void storeItemInArcadeBox(){}
+};
+
+double Speakers::combineSound(double SpeakerL, double SpeakerR)
+{
+    double combinedOutput = 0.0;
+    
+    if(monoAudio == 1) combinedOutput = 0.5 * (SpeakerL + SpeakerR);
+
+    return combinedOutput;
+}
+
+
+
+// 
+// UDT No. 9
+// 
+
+struct Coin 
+{
+    double coinDiameter = 0.0;
+    double coinWeight = 0.0;
+
+    bool inserted()
+    {
+        return true;
+    }
```

Comment:
What's the point of a function that always returns true?
- [ ] revise

Reply:
I agree, I don't need that function at all. I intended to use that as the 'action of inserting,' rather than verifying whether or not something has been inserted, if that makes sense. There's probably a better way of doing that that I have yet to explore.